app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,230 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.list
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.defaultMinSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+val LIST_ITEM_HEIGHT = 56.dp
+
+/**
+ * List item component to be used to display text with an optional second line
+ * with the flexibility to add custom UI to either end.
+ *
+ * @param label The primary text in the item.
+ * @param description An optional second line of text.
+ * @param onClick Optional lambda for handling clicks.
+ * @param beforeListAction Optional Composable for adding UI to the start of the list item.
+ * @param afterListAction Optional Composable for adding UI to the end of the list item.
+ */
+@Composable
+fun ListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    beforeListAction: @Composable RowScope.() -> Unit = {},
+    afterListAction: @Composable RowScope.() -> Unit = {},
```

Comment:
I'm open to changing the naming here. I based them on the specs
![image](https://user-images.githubusercontent.com/87384386/160928773-71bd6288-8f11-441b-b478-079065c958f9.png)

Reply:
We had a similar conversation in https://github.com/mozilla-mobile/android-components/pull/11891#discussion_r834534022 about this naming.

However, looking at the figma designs for https://www.figma.com/file/ujdiReypSypu2GhRgzArea/branch/ddGnMfYggXGmcTYfoI0aS8/Mobile-Android-Components, it seems like a list item will only have an optional icon at the beginning. So, maybe instead of taking a @Composable parameter for the before, maybe we should just be a bit more specific and only take a drawable for the optional icon. This kinda goes back to our component philosophy discussion where we want the components to not fill an 100% use case, but more specific to the intended design component.